### PR TITLE
Fix Dockerfiles

### PR DESCRIPTION
**Pull request by ybznek**
Fix atomic build dependencies

### CODE REVIEW

File: atomic/f23_spc/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ RUN dnf -y install atomic && dnf clean all
 
 # TODO: Don't clone from scratch, instead add local working copy there
 RUN dnf -y install git python3 python3-devel python3-gobject-base \
-    python3-setuptools dbus python3-dbus && dnf clean all && \
+    python3-setuptools dbus python3-dbus python-requests && dnf clean all && \
```
Comment (jan-cerny) [inline atomic/f23_spc/Dockerfile]: python-requests is not needed to build daemon. We have already tried to build f23 SPC using this dockerfile and the build was successful. I think it is only needed to build atomic. But in this Dockerfile we don't build atomic, we install atomic using dnf. Compare this with other Dockerfiles.

File: atomic/f22_spc/Dockerfile [hunk 2]
```diff
@@ -2,31 +2,9 @@
...
-    git clone https://github.com/projectatomic/atomic.git && \
-    pushd /atomic && PREFIX=/usr PYLINT=/bin/true make install && popd
-
-# TODO: Don't clone from scratch, instead add local working copy there
-RUN dnf -y install python-setuptools dbus-python && dnf clean all && \
-    git clone https://github.com/OpenSCAP/openscap-daemon && \
-    pushd /openscap-daemon && python setup.py install && popd
+RUN dnf -y install scap-security-guide openscap && dnf clean all
+
+RUN dnf -y install atomic && dnf clean all
```
Comment (jan-cerny) [inline atomic/f22_spc/Dockerfile]: Missing install openscap-utils openscap-daemon openscap-containers.
We already have openscap-containers-1.2.8 in Fedora 22.
I think that the Dockerfile for f22_spc would be very similar to f23_spc dockerfile

**Comment by jan-cerny** [comment]
Any ideas for better names of directories than `atomic` and `atomic-git`?

File: atomic-git/f22_spc/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# based on Dockerfile by Brent Baude
+
+FROM openscap-daemon-f22
+
+RUN dnf -y update && dnf -y groupinstall "Development Tools" && dnf -y install make wget git && dnf clean all
+
+RUN dnf remove -y openscap-containers && \
```
Comment (jan-cerny) [inline atomic-git/f22_spc/Dockerfile]: Is it necessary to remove the packages?

File: atomic/rhel7_spc/Dockerfile [hunk 4]
```diff
@@ -4,29 +4,9 @@ FROM rhel7
...
-    pushd /atomic && PREFIX=/usr PYLINT=/bin/true GO_MD2MAN=/bin/true \
-    make install && popd
-
-# TODO: Don't clone from scratch, instead add local working copy there
-RUN yum -y install python-setuptools dbus-python && yum clean all && \
-    git clone https://github.com/OpenSCAP/openscap-daemon && \
-    pushd /openscap-daemon && python setup.py install && popd
+RUN yum-config-manager --enable rhel-7-server-extras-rpms && yum -y install docker docker-registry atomic && yum clean all
+
+RUN yum -y install openscap-containers scap-security-guide && yum clean all
```
Comment (jan-cerny) [inline atomic/rhel7_spc/Dockerfile]: install openscap-daemon as well

File: atomic-git/f22_spc/Dockerfile (hunk 3 above)
Comment (ybznek) [inline atomic-git/f22_spc/Dockerfile]: Not necessary, but safer. If "manual" installation don't override package files, we can use wrong version to test.

**Comment by ybznek** [comment]
I will wait, until https://github.com/projectatomic/atomic/issues/302 will be fixed. We cannot build atomic in container, now (in easy way).

File: atomic-git/f22_spc/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    dnf -y install autoconf automake libtool curl-devel libxml2-devel \
```
Comment (isimluk) [inline atomic-git/f22_spc/Dockerfile]: `dnf builddep openscap && dnf install libtool`

should do the trick. The point is that all the dependencies are in the openscap.spec in Fedora. And once they change, you will get them for free.

**Comment by jan-cerny** [comment]
Please update also the README .md in `atomic` directory.

File: atomic/f22_spc/Dockerfile [hunk 6]
```diff
@@ -2,31 +2,11 @@
...
-
-# TODO: Don't clone from scratch, instead add local working copy there
-RUN dnf -y install python-setuptools dbus-python && dnf clean all && \
-    git clone https://github.com/OpenSCAP/openscap-daemon && \
-    pushd /openscap-daemon && python setup.py install && popd
+RUN dnf -y install atomic && dnf clean all
+
+RUN dnf -y install openscap-containers scap-security-guide openscap-daemon && dnf clean all
+
+RUN dnf -y install wget && dnf clean all
```
Comment (jan-cerny) [inline atomic/f22_spc/Dockerfile]: Is it possible to install all packages in one `dnf` command? If we have only one RUN instruction, the built image will have less layers.

File: atomic/f23_spc/Dockerfile [hunk 7]
```diff
@@ -2,14 +2,11 @@
...
 RUN dnf -y install atomic && dnf clean all
 
-# TODO: Don't clone from scratch, instead add local working copy there
-RUN dnf -y install git python3 python3-devel python3-gobject-base \
-    python3-setuptools dbus python3-dbus && dnf clean all && \
-    git clone https://github.com/OpenSCAP/openscap-daemon && \
-    pushd /openscap-daemon && python3 setup.py install && popd
+RUN dnf -y install openscap-containers scap-security-guide openscap-daemon && dnf clean all
+
+RUN dnf -y install wget && dnf clean all
```
Comment (jan-cerny) [inline atomic/f23_spc/Dockerfile]: Is it possible to install all packages in one dnf command? If we have only one RUN instruction, the built image will have less layers.

File: atomic/rhel7_spc/Dockerfile [hunk 8]
```diff
@@ -2,31 +2,24 @@
...
+
+RUN rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
+    yum -y install openscap-daemon && \
+    yum clean all
+
+RUN yum -y install wget && \
+    yum clean all
+
+RUN yum -y install pygobject2 && \
+    yum clean all # fix daemon dependency, TODO - remove it after ned wademon release
```
Comment (jan-cerny) [inline atomic/rhel7_spc/Dockerfile]: Please fix the typo in comment

File: atomic/rhel7_spc/Dockerfile [hunk 9]
```diff
@@ -2,31 +2,24 @@
...
-    python-docker-py libselinux-python && yum clean all && \
-    git clone https://github.com/projectatomic/atomic.git && \
-    pushd /atomic && PREFIX=/usr PYLINT=/bin/true GO_MD2MAN=/bin/true \
-    make install && popd
-
-# TODO: Don't clone from scratch, instead add local working copy there
-RUN yum -y install python-setuptools dbus-python && yum clean all && \
-    git clone https://github.com/OpenSCAP/openscap-daemon && \
-    pushd /openscap-daemon && python setup.py install && popd
+RUN yum clean all && \
```
Comment (jan-cerny) [inline atomic/rhel7_spc/Dockerfile]: Why in `yum clean all` at the beginning? Could you explain the reason in a comment, please?

File: atomic-git/f22_spc/Dockerfile (hunk 5 above)
Comment (ybznek) [inline atomic-git/f22_spc/Dockerfile]: thanks

File: atomic/f22_spc/Dockerfile (hunk 6 above)
Comment (ybznek) [inline atomic/f22_spc/Dockerfile]: We can do it, question is if it is definitely advantage.

File: atomic/rhel7_spc/Dockerfile (hunk 9 above)
Comment (ybznek) [inline atomic/rhel7_spc/Dockerfile]: Actually I don't know, but without it it doesn't work.

File: atomic/f22_spc/Dockerfile (hunk 6 above)
Comment (jan-cerny) [inline atomic/f22_spc/Dockerfile]: The more layers the worse.

File: atomic-git/rhel7_spc/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+# PYLINT=/bin/true skips pylint checks
+RUN yum remove -y atomic; yum -y install python-setuptools dbus-python docker pygobject2 \
+    python-docker-py libselinux-python python-requests && yum clean all && \
+    git clone https://github.com/projectatomic/atomic.git && \
+    pushd /atomic && PREFIX=/usr PYLINT=/bin/true GO_MD2MAN=/bin/true \
+    make install && popd
+
+RUN yum remove -y openscap-daemon; yum -y install python-setuptools dbus-python && yum clean all && \
+    git clone https://github.com/OpenSCAP/openscap-daemon && \
+    pushd /openscap-daemon && python setup.py install && popd
```
Comment (jan-cerny) [inline atomic-git/rhel7_spc/Dockerfile]: missing LABEL, CMD instructions

Comment (ybznek) [inline atomic-git/rhel7_spc/Dockerfile]: Yes, they are inherited.

File: atomic-git/f23_spc/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+RUN dnf -y install 'dnf-command(builddep)' && dnf builddep -y atomic openscap-daemon openscap && dnf -y install libtool && dnf clean all
+
+RUN git clone -b maint-1.2 https://github.com/OpenSCAP/openscap.git && \
+    pushd /openscap && ./autogen.sh && \
+    ./configure --enable-sce --prefix=/usr && make -j 4 install && popd
+
+RUN git clone https://github.com/OpenSCAP/openscap-daemon && \
+    pushd /openscap-daemon && python3 setup.py install && popd
+
+RUN dnf -y install python3-requests && git clone https://github.com/projectatomic/atomic.git && \
```
Comment (jan-cerny) [inline atomic-git/f23_spc/Dockerfile]: python3-requests should be already installed by `dnf builddep -y atomic` on line 7

File: atomic-git/f22_spc/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+RUN dnf -y install 'dnf-command(builddep)' && dnf builddep -y atomic openscap-daemon openscap && dnf -y install libtool && dnf clean all
+
+RUN git clone -b maint-1.2 https://github.com/OpenSCAP/openscap.git && \
+    pushd /openscap && ./autogen.sh && \
+    ./configure --enable-sce --prefix=/usr && make -j 4 install && popd
+
+RUN git clone https://github.com/OpenSCAP/openscap-daemon && \
+    pushd /openscap-daemon && python setup.py install && popd
+
+RUN dnf -y install python-requests && git clone https://github.com/projectatomic/atomic.git && \
```
Comment (jan-cerny) [inline atomic-git/f22_spc/Dockerfile]: python-requests should be installed by dnf builddep atomic

**Comment by jan-cerny** [comment]
@ybznek Thank you for your great contribution. Overall it looks good to me. Ack.

**Pull request merged**